src/main/java/htsjdk/beta/plugin/HtsCodec.java

```diff
@@ -0,0 +1,371 @@
+     */
+    HtsContentType getContentType();
+
+    /**
+     * Get the name of the file format supported by this codec. The format name defines the underlying
+     * format handled by this codec, and also corresponds to the format of the primary bundle
+     * resource that is required when decoding or encoding (see
+     * {@link htsjdk.beta.plugin.bundle.BundleResourceType} and {@link BundleResource#getFileFormat()}).
+     *
+     * @return the name of the underlying file format handled by this codec
+     */
+    String getFileFormat();
+
+    /**
+     * Get the version of the file format returned by {@link #getFileFormat()} that is supported by this codec.
+     *
+     * @return the file format version ({@link HtsVersion}) supported by this codec
+     */
+    HtsVersion getVersion();
+
+    /**
+     * Get a user-friendly display name for this codec.
+     * </p>
+     * It is recommended that the display name minimally include both the name of the supported file format and
+     * the supported version.
+     *
+     * @return a user-friendly display name for this codec
+     */
+    default String getDisplayName() {
+        return String.format("%s/%s/%s", getFileFormat(), getVersion(), getClass().getName());
+    }
+
+    /**
+     * Determine if this codec "owns" the URI contained in {@code ioPath} see ({@link IOPath#getURI()}).
+     * <p>
+     * A codec "owns" the URI only if it has specific requirements on the URI protocol scheme, URI format,
+     * or query parameters that go beyond a simple file extension, AND it explicitly recognizes the URI
+     * as conforming to those requirements. File formats that only require a specific file extension should
+     * always return false from {@link #ownsURI}, and should instead use the extension as a filter in
+     * {@link #canDecodeURI(IOPath)}.
+     * <p>
+     * Returning true from this method will cause the framework to bypass the stream-oriented signature
+     * probing that is used to resolve inputs to a codec handler. During codec resolution, if any registered
+     * codec returns true for this method on {@code ioPath}, the signature probing protocol will instead:
+     * <ol>
+     * <li> immediately prune the list of candidate codecs to only those that return true for this method
+     * on {@code ioPath} </li>
+     * <li> not attempt to obtain an InputStream on the IOPath containing the URI, on the assumption that
+     * special handling is required in order to access the underlying resource (i.e., htsget
+     * codec would claim an "http://" URI if the rest of the URI conforms to the expected format for that
+     * codec's protocol). </li>
+     * </ol>
+     * <p>
+     * Any codec that returns true from {@link #ownsURI(IOPath)} for a given IOPath must also return true
+     * from {@link #canDecodeURI(IOPath)} for the same IOPath.
+     *
+     * For custom URI handlers, codecs should avoid making remote calls to determine the suitability
```

Comment:
I'll try to update this a bit, but the (codec resolution) process is only about finding the correct codec/handler. Saying you own/handle the URI doesn't imply any guaranty that retrieving data from it will succeed any more than it does for a local file. Many things can fail downstream of that, and the codec can handle or report those errors when they happen. I'll try to clarify that here.